src/dictionary/api.ts

```diff
@@ -80,16 +80,26 @@ export const get = async (req: Request, res: Response) => {
   return res.status(200).send(schema);
 };
 
+export const getClinicalSchemas = async (withFields: boolean) =>
+  withFields
+    ? await manager
+        .instance()
+        .getSchemasWithFields()
+        .then(result =>
+          result.filter(
+            (s: manager.SchemaWithFields) => s.name !== ClinicalEntitySchemaNames.REGISTRATION,
+          ),
+        )
+    : await manager
+        .instance()
+        .getSchemaNames()
+        .then(result => result.filter(s => s !== ClinicalEntitySchemaNames.REGISTRATION));
+
```

Comment:
is this logic the same apart from the instance method call?

could be little more DRY. not sure which is the async call so not sure where to put await..
``` javascript
export const getClinicalSchemas = async (withFields: boolean) => {
  const managerInstance = await manager.instance();
  const schemaNameFilter =  (schemaName) => schemaName !== ClinicalEntitySchemaNames.REGISTRATION;
  const schemas = withFields ? managerInstance.getSchemasWithFields().map(schema => schema.name) : managerInstance.getSchemaNames();
  return schemas.filter(schemaNameFilter)
}
```

Reply:
Yes I had a feeling this could be improved. But, we don't want to map `getSchemasWithFields()`, we need to return the objects with both names + fields, so `schemaNameFilter` needs to be called differently depending on value of withFields: https://github.com/icgc-argo/argo-clinical/pull/1060/commits/67ceec9e941a99262523a729233d3a15437078b5